jbmc/unit/java-testing-utils/require_type.cpp

```diff
@@ -54,40 +54,40 @@ struct_union_typet::componentt require_type::require_component(
 
 /// Checks a type is a code_type (i.e. a function)
 /// \param type: The type to check
-/// \return The cast version of the type code_type
-code_typet require_type::require_code(const typet &type)
+/// \return The cast version of the type method_type
+java_method_typet require_type::require_code(const typet &type)
 {
   REQUIRE(type.id() == ID_code);
```

Comment:
If you introduce the following in java_types:

```
template <>
inline bool can_cast_type<java_method_typet>(const typet &type)
{
  return type.id() == ID_code && type.get_bool(ID_C_java_method_type);
}
```
Then you can use that here instead of checking the id directly (and will allow you to also check the bool. This only applies if you end up with a `require_java_type` method.